src/main/java/org/opensearch/agent/tools/SearchIndexTool.java

```diff
@@ -96,16 +96,10 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
             SearchRequest searchRequest;
             try {
```

Comment:
Remove this try...catch block since the outer try...catch can catch the exception here.

Reply:
This try catch block is needed because we need to try another different parsing method.